templates/default/conf.d/90-quota.conf.erb

```diff
@@ -7,8 +7,16 @@
 # Note that you also have to enable quota plugin in mail_plugins setting.
 # <doc/wiki/Quota.txt>
 
-<% if @plugins.has_key?('quota') and @plugins['quota'].kind_of?(Hash) -%>
+<% if @plugins.has_key?('quota') or @services['quota-status'].kind_of?(Hash) or @services['quota-warning'].kind_of?(Hash) -%>
+  <% if @plugins['quota'].kind_of?(Hash) -%>
 <%=  Dovecot::Conf.plugin('quota', @plugins['quota']) %>
+  <% end -%>
+  <% if @services['quota-status'].length > 0 -%>
+<%=  Dovecot::Conf.service('quota-status', @services['quota-status']) %>
+  <% end -%>
+  <% if @services['quota-warning'].length > 0 -%>
+<%=  Dovecot::Conf.service('quota-warning', @services['quota-warning']) %>
+  <% end -%>
```

Comment:
I have pushed my second go. Checking empty?, kind_of? or anything else on NilClass will throw error. I'm not rubyist so I hope my second go is ok :)